sdk/mediaservices/media-analytics-edge/review/media-analytics-edge.api.md

```diff
@@ -0,0 +1,523 @@
+    "@type": "#Microsoft.Media.MediaGraphSignalGateProcessor";
+    activationEvaluationWindow?: string;
+    activationSignalOffset?: string;
+    minimumActivationTime?: string;
+    maximumActivationTime?: string;
+};
+
+// @public
+export interface MediaGraphSink {
+    "@type": "#Microsoft.Media.MediaGraphIoTHubMessageSink" | "#Microsoft.Media.MediaGraphFileSink" | "#Microsoft.Media.MediaGraphAssetSink";
+    inputs: MediaGraphNodeInput[];
+    name: string;
+}
+
+// @public (undocumented)
+export type MediaGraphSinkUnion = MediaGraphSink | MediaGraphIoTHubMessageSink | MediaGraphFileSink | MediaGraphAssetSink;
+
+// @public
+export interface MediaGraphSource {
+    "@type": "#Microsoft.Media.MediaGraphRtspSource" | "#Microsoft.Media.MediaGraphIoTHubMessageSource";
+    name: string;
+}
+
+// @public (undocumented)
+export type MediaGraphSourceUnion = MediaGraphSource | MediaGraphRtspSource | MediaGraphIoTHubMessageSource;
+
+// @public
+export interface MediaGraphSystemData {
+    createdAt?: Date;
+    lastModifiedAt?: Date;
+}
+
+// @public
+export type MediaGraphTlsEndpoint = MediaGraphEndpoint & {
+    "@type": "#Microsoft.Media.MediaGraphTlsEndpoint";
+    trustedCertificates?: MediaGraphCertificateSourceUnion;
+    validationOptions?: MediaGraphTlsValidationOptions;
+};
+
+// @public
+export interface MediaGraphTlsValidationOptions {
+    ignoreHostname?: string;
+    ignoreSignature?: string;
+}
+
+// @public
+export interface MediaGraphTopology {
+    name: string;
+    properties?: MediaGraphTopologyProperties;
+    systemData?: MediaGraphSystemData;
+}
+
+// @public
+export type MediaGraphTopologyDeleteRequest = ItemNonSetRequestBase & {
+    methodName: "GraphTopologyDelete";
+};
+
+// @public
+export type MediaGraphTopologyGetRequest = ItemNonSetRequestBase & {
+    methodName: "GraphTopologyGet";
+};
+
+// @public
+export type MediaGraphTopologyListRequest = MethodRequest_2 & {
+    methodName: "GraphTopologyList";
+};
+
+// @public
+export interface MediaGraphTopologyProperties {
+    description?: string;
+    parameters?: MediaGraphParameterDeclaration[];
+    processors?: MediaGraphProcessorUnion[];
+    sinks?: MediaGraphSinkUnion[];
+    sources?: MediaGraphSourceUnion[];
+}
+
+// @public
+export type MediaGraphTopologySetRequest = MethodRequest_2 & {
+    methodName: "GraphTopologySet";
+    graph: MediaGraphTopology;
+};
+
+// @public
+export type MediaGraphTopologySetRequestBody = MethodRequest_2 & MediaGraphTopology & {
+    methodName: "MediaGraphTopologySetRequestBody";
+};
+
+// @public
+export type MediaGraphUnsecuredEndpoint = MediaGraphEndpoint & {
+    "@type": "#Microsoft.Media.MediaGraphUnsecuredEndpoint";
+};
+
+// @public
+export type MediaGraphUsernamePasswordCredentials = MediaGraphCredentials & {
+    "@type": "#Microsoft.Media.MediaGraphUsernamePasswordCredentials";
+    username: string;
+    password: string;
+};
+
+// @public (undocumented)
+export interface MethodRequest {
```

Comment:
This interface feels extremely bizarre to me, like you're doing RPC through a REST wrapper?

Reply:
I added a PR description that explains our sdk a bit better. Essentially, there is no client in this sdk. We only want the user to be able to create JSON objects and then pass those objects to the Azure IoT Hub sdk. 

We had a previous version of this sdk that had a client that called the Azure IoT Hub sdk directly but we did not want to take a dependency on their sdk as the user might want to use a different version of it so we settled on this models only sdk pattern.

Reply:
This is interesting. Since it's a complete divergence from the client model we follow in Track2 libraries, I think we'd have to get special board approval for it.

Do you find the methods you export provide much? Or could you turn this into a types-only package?

/cc @bterlson